ops/charm.py

```diff
@@ -402,23 +402,32 @@ class RelationDepartedEvent(RelationEvent):
 
     Attributes:
         departing_unit: The :class:`~ops.model.Unit` that is departing.  This
-            can facilitate determining e.g. whether *you* are the departing unit.
+            can facilitate determining e.g. whether *you* are the departing
+            unit.
     """
 
-    def __init__(self, handle, relation, app=None, unit=None, departing_unit=None):
+    def __init__(self, handle, relation, app=None, unit=None,
+                 departing_unit_name=None):
         super().__init__(handle, relation, app=app, unit=unit)
 
-        self.departing_unit = None
-        if departing_unit:
-            self.departing_unit = self.framework.model.get_unit(departing_unit)
+        self._departing_unit_name = departing_unit_name
+
+    @property
+    def departing_unit(self) -> typing.Optional[model.Unit]:
+        """The `ops.model.Unit` that is departing, if any."""
+        # doing this on init would fail because `framework` gets patched in
+        # post-init
+        if not self._departing_unit_name:
+            return None
+        return self.framework.model.get_unit(self._departing_unit_name)
```

Comment:
that is not the point, see the comment. The reason for this design is that `framework` is not known at init time. We need to be lazy with departing_unit because of that.